### PR TITLE
Dev/zahalzel/b2c updates

### PR DESCRIPTION
**Pull request by zahalzel**
Adding support for configuring app registrations that exist in Azure AD B2C tenants

### CODE REVIEW

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 1]
```diff
@@ -347,32 +356,30 @@ private static string UpdateCallbackPath(string redirectUri, bool isBlazorWasm =
...
-        internal static bool UpdateImplicitGrantSettings(Application app, ProvisioningToolOptions toolOptions)
+        internal static bool UpdateImplicitGrantSettings(Application app, ProvisioningToolOptions toolOptions, bool isB2C = false)
         {
             bool needsUpdate = false;
             var currentSettings = app.Web.ImplicitGrantSettings;
 
-            if (toolOptions.IsBlazorWasm) // In the case of Blazor WASM, Access Tokens and Id Tokens must both be true.
+            // In the case of Blazor WASM and B2C, Access Tokens and Id Tokens must both be true.
+            if ((toolOptions.IsBlazorWasm || isB2C)
+                && currentSettings.EnableAccessTokenIssuance is false
```
Comment (zahalzel) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:366]: Fix this conditional

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 2]
```diff
@@ -188,6 +188,12 @@ public class MicrosoftIdentityPlatformApplicationManager
 
             var graphServiceClient = GetGraphServiceClient(tokenCredential);
 
+
+            // TODO: Add if it's B2C, acquire the SUSI Policy
```
Comment (zahalzel) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:192]: Will do in separate PR

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/Tool/ProvisioningToolOptions.cs [hunk 3]
```diff
@@ -173,7 +173,7 @@ public string ProjectTypeIdentifier
         /// <summary>
         /// Determines if the project type is blazor wasm
         /// </summary>
-        public bool IsBlazorWasm => "blazorwasm".Equals(ProjectType) || "blazorwasm-client".Equals(ProjectType);
+        public bool IsBlazorWasm => ProjectTypes.BlazorWasm.Equals(ProjectType) || ProjectTypes.BlazorWasmClient.Equals(ProjectType);
```
Comment (zahalzel) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/Tool/ProvisioningToolOptions.cs:176]: String.Equals()

**Review by deepchoudhery — APPROVED** [review]
LGTM
